### PR TITLE
fix(ci/firebase) Always push to firebase. Even on failure

### PR DESCRIPTION
**Pull request by talkhasib**
Signed-off-by: Tariq Al-Khasib <talkhasib@fb.com>

## Summary

We need to push failed builds to Firebase database for visibility and tracking

## Test Plan

testing on ci_firebase_always branch

### CODE REVIEW

File: .github/workflows/build_all.yml [hunk 1]
```diff
@@ -665,7 +666,7 @@ jobs:
           SLACK_FOOTER: ' '
   Publish_to_firebase:
     name: Publish to firebase
-    if: github.event_name == 'push'
+    if: always() && github.event_name == 'push'
     runs-on: ubuntu-latest
     needs: [agw-build]
```
Comment (quentinDERORY) [inline .github/workflows/build_all.yml:670]: remove the needs line here

File: .github/workflows/build_all.yml [hunk 2]
```diff
@@ -665,7 +666,7 @@ jobs:
           SLACK_FOOTER: ' '
   Publish_to_firebase:
     name: Publish to firebase
-    if: github.event_name == 'push'
+    if: always() && github.event_name == 'push'
```
Comment (quentinDERORY) [inline .github/workflows/build_all.yml:668]: don't use always here

**Review by quentinDERORY — CHANGES_REQUESTED** [review]
To push on failure you need to remove the needs link with agw-build that's all

File: .github/workflows/build_all.yml (hunk 2 above)
Comment (talkhasib) [reply .github/workflows/build_all.yml:668]: I need it to wait for the AGW job. But always work regardless of exit state.

**Review by quentinDERORY — APPROVED** [review]
just remove the added branch trigger :)

**Pull request merged**